src/runtime/graph/graph_runtime.cc

```diff
@@ -103,27 +103,82 @@ class GraphRuntime : public ModuleNode {
   void SetInput(int index, DLTensor* data_in) {
     CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
     uint32_t eid = this->entry_id(input_nodes_[index], 0);
-    TVM_CCALL(TVMArrayCopyFromTo(data_in, &data_entry_[eid], nullptr));
+    data_entry_[eid].CopyFrom(data_in);
+  }
+  /*!
+   * \brief Get the number of inputs
+   *
+   * \return The number of inputs to the graph.
+   */
+  int GetInputCount(void) {
+    return input_nodes_.size();
+  }
+  /*!
+   * \brief Get the number of outputs
+   *
+   * \return The number of outputs from graph.
+   */
+  int GetOutputCount(void) {
+    return outputs_.size();
+  }
+  /*!
+   * \brief Return NDArray for given input index.
+   * \param index The input index.
+   *
+   * \return NDArray corresponding to given input node index.
+   */
+  NDArray GetInputAsNDArray(int index) {
+    CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
+    uint32_t eid = this->entry_id(input_nodes_[index], 0);
+    return NDArray(data_entry_[eid]);
```

Comment:
This looks like copy right? You should use NDArray::CreateView()

Reply:
This is a constructor to just increment ref count, no copy here.

CreateView is required to view underlying data as different shape and dtype, which is not required in this case.

Reply:
that seems right. Can you fix the doc [here](https://github.com/dmlc/tvm/blob/master/include/tvm/runtime/ndarray.h#L34)? It is confusing.